src/Persistent/Implementation.hs

```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE CPP #-}
```

Comment:
I guess this shouldn't be too hard to make the implementation dynamically chosen we could even hide the Persist type completely I think.